linter/src/main/java/linter/App.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package linter;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class App {
+    public static void main(String[] args) {
+        String filePath = "src/main/java/linter/resources/gates.js";
+        ArrayList<String> elist = findMissingSemiColons(filePath);
+        for(String error : elist){
+            System.out.println(error);
+        }
+    }
+    public static ArrayList<String> findMissingSemiColons(String path){
+        ArrayList<String> errorList = new ArrayList<>();
+        int lineNum = 0;
+
+        try
+        {
+            Scanner scanner = new Scanner(new File(path));
+            while (scanner.hasNextLine()){
+                String line = scanner.nextLine();
+                lineNum = lineNum +1;
+                if(line.length() > 0 && !line.contains("{") && !line.contains("}") && !line.contains("if") && !line.contains("else") && !line.contains(";")){
+                    errorList.add("Line " + (lineNum) + ": Missing semicolon.");
+                }
+            }
+        }
```

Comment:
Also a C# thing to do

`} catch {`

I don't know for sure, but I assume Amazon will expect you to do thing the 'Java' way so it might be a good idea to start practicing this style. I'm never going to deduct points for this (unless I'm specifically told otherwise, in which case I'll let you know)